featuretests/storage_test.go

```diff
@@ -272,85 +298,105 @@ func (s *cmdStorageSuite) registerTmpProviderType(c *gc.C) {
 
 func (s *cmdStorageSuite) TestListPoolsProviderNoMatch(c *gc.C) {
 	s.registerTmpProviderType(c)
-	context := runPoolList(c, "--provider", string(provider.TmpfsProviderType))
+	stdout, _, err := runPoolList(c, "--provider", string(provider.TmpfsProviderType))
+	c.Assert(err, jc.ErrorIsNil)
 	expected := `
 tmpfs:
   provider: tmpfs
 `[1:]
-	c.Assert(testing.Stdout(context), gc.Equals, expected)
+	c.Assert(stdout, gc.Equals, expected)
 }
 
 func (s *cmdStorageSuite) TestListPoolsProviderUnregistered(c *gc.C) {
-	_, err := testing.RunCommand(c, envcmd.Wrap(&cmdstorage.PoolListCommand{}), "--provider", "oops")
-	c.Assert(errors.Cause(err), gc.ErrorMatches, ".*not supported.*")
+	_, stderr, err := runPoolList(c, "--provider", "oops")
+	c.Assert(err, gc.NotNil)
+	c.Assert(stderr, jc.Contains, `"oops" for environment "dummyenv" not supported`)
 }
 
 func (s *cmdStorageSuite) TestListPoolsNameAndProvider(c *gc.C) {
-	context := runPoolList(c, "--name", "block", "--provider", "loop")
+	stdout, _, err := runPoolList(c, "--name", "block", "--provider", "loop")
+	c.Assert(err, jc.ErrorIsNil)
 	expected := `
 block:
   provider: loop
   attrs:
     it: works
 `[1:]
-	c.Assert(testing.Stdout(context), gc.Equals, expected)
+	c.Assert(stdout, gc.Equals, expected)
 }
 
 func (s *cmdStorageSuite) TestListPoolsProviderAndNotName(c *gc.C) {
-	context := runPoolList(c, "--name", "fluff", "--provider", "ebs")
+	stdout, _, err := runPoolList(c, "--name", "fluff", "--provider", "ebs")
+	c.Assert(err, jc.ErrorIsNil)
 	// there is no pool that matches this name AND type
-	c.Assert(testing.Stdout(context), gc.Equals, "")
+	c.Assert(stdout, gc.Equals, "")
 }
 
 func (s *cmdStorageSuite) TestListPoolsNameAndNotProvider(c *gc.C) {
 	s.registerTmpProviderType(c)
-	context := runPoolList(c, "--name", "block", "--provider", string(provider.TmpfsProviderType))
+	stdout, _, err := runPoolList(c, "--name", "block", "--provider", string(provider.TmpfsProviderType))
+	c.Assert(err, jc.ErrorIsNil)
 	// no pool matches this name and this provider
-	c.Assert(testing.Stdout(context), gc.Equals, "")
+	c.Assert(stdout, gc.Equals, "")
 }
 
 func (s *cmdStorageSuite) TestListPoolsNotNameAndNotProvider(c *gc.C) {
 	s.registerTmpProviderType(c)
-	context := runPoolList(c, "--name", "fluff", "--provider", string(provider.TmpfsProviderType))
-	c.Assert(testing.Stdout(context), gc.Equals, "")
+	stdout, _, err := runPoolList(c, "--name", "fluff", "--provider", string(provider.TmpfsProviderType))
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(stdout, gc.Equals, "")
 }
 
-func runPoolCreate(c *gc.C, args ...string) *cmd.Context {
-	context, err := testing.RunCommand(c, envcmd.Wrap(&cmdstorage.PoolCreateCommand{}), args...)
-	c.Assert(err, jc.ErrorIsNil)
-	return context
+func runPoolCreate(c *gc.C, args ...string) (string, string, error) {
```

Comment:
because all tests will eventually want to extract StdOut, StdErr and check the error.. so might as well do it all in one place..